### ISSUE TITLE
UAC breaks touch interaction.

### ISSUE THREAD

**Issue by jticket1024** [nvaccess/nvda#13591]
### Steps to reproduce:
Disable UAC
### Actual behavior:
Touch interaction support stops working.
### Expected behavior:
Touch interaction should work either way.
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
2021.3.5
#### Windows version:
Windows 11
#### Does the issue still occur after restarting your computer?
Yes
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by lukaszgo1** [comment]
@jticket1024 Please set your log level to debug in general settings, open Python console with NVDA -Ctrl+_Z and paste the following exactly as shown:
```
import touchHandler
touchHandler.touchSupported(True)
```
After that upload your NVDA log here - be ware that at the debug level it contains all your key presses and all content spoken by NVDA so you may want to trim it.

**Comment by seanbudd** [comment]
@lukaszgo1 - This user is noting that NVDA's touch interaction is disabled on secure screens, such as the UAC dialog when running an application as administrator. I believe this is a feature of Windows, and there's nothing we can do about it.

You cannot open the python console on secure screens or capture debug logging without setting the [serviceDebug](https://www.nvaccess.org/files/nvda/documentation/userGuide.html#SystemWideParameters) parameter.

**Comment by michaelDCurran** [comment]
@jticket1024  I think we need some clarification here.
I think what you mean is that touch no longer works if you set your UAC security level to 0 in Windows, so that elevating to admin no longer asks for permission before doing so.
@seanbudd  to my knowledge, touch interaction and audio ducking should function perfectly fine on secure screens, as NVDA is being run as system. If this is not the case, there is either an unknown bug in Windows or NVDA. 
Certainly touch interaction was expected to / must work on the Windows logon screen / User Account Control dialog.

**Comment by seanbudd** [comment]
Strangely, after setting the serviceDebug parameter, touch interaction works as expected in secure screens.

**Comment by CyrilleB79** [comment]
When resolving this issue, #13593 should also be reverted or at least fixed.

**Comment by jticket1024** [comment]
Yes, when setting UAC to zero, touch interaction stops working everywhere.

**Comment by lukaszgo1** [comment]
@jticket1024 What about the log I asked you to collect in my last  comment?

**Comment by jticket1024** [comment]
I've pasted that in, and here's what I get. I've only copied the relevant bit that happens after I run that. Nothing else about touch is said. When UAC is turned on and enabled, touch interaction works just fine.
The only thing in the log when UAC is set to 0 is:
DEBUGWARNING - touchHandler.touchSupported (09:15:03.994) - MainThread (5284):
NVDA doesn't have UI Access so touch isn't supported.

**Comment by michaelDCurran** [comment]
I believe that `systemUtils.hasUIAccess()` returns false if UAC is set to 0, because our process token does not seem to get TokenUIAccess if UAC is set to 0. 
To fix this, we will need to find a way of detecting that we already have admin / a high integrity level and don't actually require UIAccess.

**Comment by gexgd0419** [comment]
Seems that UIAccess and admin rights are independent in this case. A process can be run with UIAccess but without admin rights, or without UIAccess but with admin rights, or with both, or with neither.

`RegisterPointerInputTarget` returns `ERROR_ACCESS_DENIED` when running with admin rights but without UIAccess, so I guess that detecting admin rights does not work here.

However, this does work for audio ducking (issue #15790), as audio ducking works with either admin rights or UIAccess.
